raiden-ts/src/types.ts

```diff
@@ -49,3 +49,15 @@ export interface RaidenEpicDeps {
   userDepositContract: UserDeposit;
   main?: { signer: Signer; address: Address };
 }
+
+export interface ChangeEvent<T, P> {
+  readonly type: T;
+  readonly payload: P;
+}
+
+export type OnChange<T, P> = (event: ChangeEvent<T, P>) => void;
+
+export enum Deposit {
```

Comment:
Maybe it'd be useful to name it not only an enum for Deposit steps, but for overall callback events types, then it'd be already a beginning for general callbacks for the other multistep operations as well